smack-core/src/main/java/org/jivesoftware/smack/packet/IQ.java

```diff
@@ -117,6 +117,7 @@ public final XmlStringBuilder toXML() {
         XmlStringBuilder buf = new XmlStringBuilder();
         buf.halfOpenElement(IQ_ELEMENT);
         addCommonAttributes(buf);
+        addCustomAttributes(buf);
```

Comment:
I would merge `addCommonattributes` and `addCustomAttributes`, so just add the code in `addCustomAttributes` to the already existing method and rename it to `addAttributes`.